src/libraries/Common/src/System/Security/Cryptography/RSASecurityTransforms.cs

```diff
@@ -496,19 +492,29 @@ public override bool VerifyHash(ReadOnlySpan<byte> hash, ReadOnlySpan<byte> sign
 
                 ThrowIfDisposed();
 
-                if (padding == RSASignaturePadding.Pkcs1)
+                Interop.AppleCrypto.PAL_HashAlgorithm palAlgId = PalAlgorithmFromAlgorithmName(hashAlgorithm, out _);
+                Interop.AppleCrypto.PAL_SignatureAlgorithm signatureAlgorithm = padding.Mode switch
+                {
+                    RSASignaturePaddingMode.Pss => Interop.AppleCrypto.PAL_SignatureAlgorithm.RsaPss,
+                    RSASignaturePaddingMode.Pkcs1 => Interop.AppleCrypto.PAL_SignatureAlgorithm.RsaPkcs1,
+                    _ => throw new CryptographicException(SR.Cryptography_InvalidPaddingMode)
+                };
+
+                // Apple's native implementation does not support RSA+PSS with MD5. We use raw RSA padding and perform
+                // the padding manually in that circumstance.
+                if (signatureAlgorithm != Interop.AppleCrypto.PAL_SignatureAlgorithm.RsaPss ||
+                    palAlgId != Interop.AppleCrypto.PAL_HashAlgorithm.Md5)
                 {
-                    Interop.AppleCrypto.PAL_HashAlgorithm palAlgId =
-                        PalAlgorithmFromAlgorithmName(hashAlgorithm, out _);
                     return Interop.AppleCrypto.VerifySignature(
                         GetKeys().PublicKey,
                         hash,
                         signature,
                         palAlgId,
-                        Interop.AppleCrypto.PAL_SignatureAlgorithm.RsaPkcs1);
+                        signatureAlgorithm);
                 }
-                else if (padding.Mode == RSASignaturePaddingMode.Pss)
+                else
```

Comment:
Nit: remove the else and unindent the block's contents?

Reply:
Yeah. That looks nicer.